### PR TITLE
fix required fields (asterisc) inside repeatable

### PR DESCRIPTION
**Pull request by pxpm**
### BEFORE - What was wrong? What was happening before this PR?

The repeatable fields could not properly get the astheriscs!

This PR needs another one in PRO to properly work.

### AFTER - What is happening after this PR?

It correctly displays the asteriscs when fields are required.

### How did you achieve that, in technical terms?

fixed the way we handle subfields when required.

### Is it a breaking change or non-breaking change?
non

### How can we test the before & after?

add some repeatable fields with `required` in the form request `repeatable.*.field => 'required'`. No asterics for required fields. After the PR, they have the asteriscs!

### CODE REVIEW

File: src/app/Library/CrudPanel/Traits/Validation.php [hunk 1]
```diff
@@ -191,8 +191,7 @@ public function setRequiredFields($classOrRulesArray)
                 }
             }
         }
-
-        $this->setOperationSetting('requiredFields', $requiredFields);
+        $this->setOperationSetting('requiredFields', array_unique(array_merge($this->getOperationSetting('requiredFields') ?? [], $requiredFields)));
```
Comment (tabacitu) [inline src/app/Library/CrudPanel/Traits/Validation.php:194]: Ugh... such a long line. Let's split it up so it's easy to understand what's happening. Which is also a sine that this needs a comment, explaining why this is needed.

```suggestion
        $requiredFields = array_merge($this->getOperationSetting('requiredFields') ?? [], $requiredFields);
        $requiredFields = array_unique($requiredFields);

        $this->setOperationSetting('requiredFields', $requiredFields);
```

File: src/resources/views/crud/fields/inc/wrapper_start.blade.php [hunk 2]
```diff
@@ -6,15 +6,18 @@
...
     }
-	// if the field is required in the FormRequest, it should have an asterisk.
-	// we add the base entity to the field name to account for nested relation fields validated with `field.*.key`
-	$fieldName = isset($field['baseEntity']) ? $field['baseEntity'].'.'.$field['name'] : $field['name'];
-	$fieldName = is_array($fieldName) ? current($fieldName) : $fieldName;
-	$required = (isset($action) && $crud->isRequired($fieldName)) ? ' required' : '';
+	// if the field is required in any of the crud validatiors (FormRequest, Controller or Field validation) 
+	// we add an astherisc for it. Case it's a subfield, that check is done upstream in repeatable_row. 
+	// the reason for that is that here the field name is already the repeatable name: parent[row][fieldName]
+	if(!isset($field['parentFieldName'])) {
```
Comment (tabacitu) [inline src/resources/views/crud/fields/inc/wrapper_start.blade.php:12]: Wait, what's this? When did this show up? I've searched both CRUD and PRO for `parentFieldName` and couldn't find anything.

**Review by tabacitu — CHANGES_REQUESTED** [review]

**Comment by tabacitu** [comment]
Also reviewed https://github.com/DigitallyHappy/backpack-pro/pull/10

File: src/resources/views/crud/fields/inc/wrapper_start.blade.php (hunk 2 above)
Comment (pxpm) [reply src/resources/views/crud/fields/inc/wrapper_start.blade.php:12]: totally rigth! 

That's how I started, but then removed it as we have `baseEntity` or `baseModel` to deal with subfields! Fixing! \o

**Pull request merged**